common/beerocks/scripts/prplmesh_utils.sh.in

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+
+BWL_TYPE=@BWL_TYPE@
+BTL_TYPE=@BTL_TYPE@
+
+dbg() {
+    [ "$VERBOSE" == "true" ] && echo "$@"
+}
+
+err() {
+	echo -e '\033[1;31m'"$@"'\033[0m'
+}
+
+run() {
+    dbg "$*"
+    "$@" || exit $?
+}
+
+platform_init_dummy() {
+    echo "platform dummy mode init..."
+    run ip link add wlan0 type dummy
+    run ip link add wlan2 type dummy
+    run ip link add eth1 type dummy
+    run brctl addbr br-lan
+    run brctl addif br-lan eth1
+    run brctl addif br-lan wlan0
+    run brctl addif br-lan wlan2
+    run ip link set eth1 up
+    run ip link set wlan0 up
+    run ip link set wlan2 up
+    run ip link set br-lan up
+    echo "Done"
+}
+
+platform_deinit_dummy() {
+    echo "platform dummy mode de-init..."
+    ip link set br-lan down
+    brctl delbr br-lan
+    ip link set eth1 down
+    ip link set wlan0 down
+    ip link set wlan2 down
+    ip link del eth1
+    ip link del wlan0
+    ip link del wlan2
+    echo "Done"
+}
+
+prplmesh_framework_init() {
+    echo "prplmesh_framework_init - starting local_bus and ieee1905_transport processes..."
+    @CMAKE_INSTALL_PREFIX@/bin/local_bus &
+    @CMAKE_INSTALL_PREFIX@/bin/ieee1905_transport &
+}
+
+prplmesh_framework_deinit() {
+    echo "prplmesh_framework_init - killing local_bus and ieee1905_transport processes..."
+    killall local_bus ieee1905_transport
+}
+
+prplmesh_controller_start() {
+    echo "prplmesh_controller_start - start beerocks_controller process..."
+    @CMAKE_INSTALL_PREFIX@/bin/beerocks_controller &
+}
+
+prplmesh_controller_stop() {
+    echo "prplmesh_controller_stop - stopping beerocks_controller process..."
+    killall beerocks_controller
+}
+
+prplmesh_agent_start() {
+    echo "prplmesh_agent_start - start beerocks_agent process..."
+    @CMAKE_INSTALL_PREFIX@/bin/beerocks_agent &
+}
+
+prplmesh_agent_stop() {
+    echo "prplmesh_agent_stop - stopping beerocks_agent process..."
+    killall beerocks_agent
+}
+
+start_function() {
+    echo "$0: start"
+    if [ `id -u` -ne 0 ]; then
+        err "$0: This command must be run as root"
+        exit 1
+    fi
+    [[ "$BWL_TYPE" == "DUMMY" ]] && platform_init_dummy
+    [[ "$BTL_TYPE" == "LOCAL_BUS" ]] && prplmesh_framework_init
+    [[ "$PRPLMESH_MODE" == "CA" || "$PRPLMESH_MODE" == "C" ]] && prplmesh_controller_start
+    [[ "$PRPLMESH_MODE" == "CA" || "$PRPLMESH_MODE" == "A" ]] && prplmesh_agent_start
+}
+
+stop_function() {
+    echo "$0: stop"
+    if [ `id -u` -ne 0 ]; then
+        err "$0: This script must be run as root"
+        exit 1
+    fi
+    [[ "$BWL_TYPE" == "DUMMY" ]] && platform_deinit_dummy
```

Comment:
Just for my information: why use `[[` instead of `[`?

Reply:
just more features provided by `bash` - see [here](https://stackoverflow.com/questions/3427872/whats-the-difference-between-and-in-bash)

Reply:
For the record: the second answer of [another stackoverflow question](https://stackoverflow.com/questions/669452/is-double-square-brackets-preferable-over-single-square-brackets-in-ba) advises against using `[[`:
> Recommendation: always use [].
>
> There are POSIX equivalents for every [[ ]] construct I've seen.
>
> If you use [[ ]] you:
>
> * lose portability
> * force the reader to learn the intricacies of another bash extension. [ is just a regular command with a weird name, no special semantics are involved.

I'm fine either way though. We should just try to be consistent.

Reply:
convinced me, I'll change to `[`